File: simplified-cardcreator/src/main/java/org/nypl/simplified/cardcreator/model/Address.java
package org.nypl.simplified.cardcreator.model;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.Serializable;

/**
 * Created by aferditamuriqi on 8/29/16.
 *
 */

public class Address  implements Serializable {


    public String line_1;
    public String line_2;
    public String city;
    public String county;
    public String state;
    public String zip;
    public Boolean is_residential;

    public Address(JSONObject address) {


        try {
            line_1 = address.getString("line_1");
            line_2 = address.getString("line_2");
            city = address.getString("city");
            county = address.getString("county");
            state = address.getString("state");
            zip = address.getString("zip");
            Object obj = address.get("is_residential");
            if (obj instanceof Boolean) {
                is_residential = address.getBoolean("is_residential");
            }
        } catch (JSONException e) {
            e.printStackTrace();
        }

    }
}

File: simplified-cardcreator/src/main/java/org/nypl/simplified/cardcreator/model/Address.kt
package org.nypl.simplified.cardcreator.model

import com.squareup.moshi.Json
import org.nypl.simplified.cardcreator.utils.checkFieldNotNull

data class Address(
  @field:Json(name = "line1") val line1: String,
  @field:Json(name = "line2") val line2: String = "",
  @field:Json(name = "city") val city: String,
  @field:Json(name = "state") val state: String,
  @field:Json(name = "zip") val zip: String,
  @field:Json(name = "isResidential") val isResidential: Boolean,
  @field:Json(name = "hasBeenValidated") val hasBeenValidated: Boolean = true
) {

  /**
   * Moshi might have set some missing fields to null instead of throwing an exception,
   * so we need to manually check every field is not null to prevent NullPointerExceptions where
   * no-one would expect them.
   */

  fun validate(): Address =
    this.apply {
      checkFieldNotNull(line1, "line1")
      checkFieldNotNull(line2, "line2")
      checkFieldNotNull(city, "city")
      checkFieldNotNull(state, "state")
      checkFieldNotNull(zip, "zip")
      checkFieldNotNull(isResidential, "isResidential")
      checkFieldNotNull(hasBeenValidated, "hasBeenValidated")
    }
}
